src/pages/home-page/home-page.js

```diff
@@ -1,49 +1,46 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Header, VideoThumbnail, ErrorMessage } from '../../components';
+import { propTypes } from './prop-types';
 import { StyledWrapper } from './styled';
 import { fetchTopVideos } from '../../api';
 
-class HomePage extends React.Component {
-  state = {
-    topVideos: { items: [], error: false },
-  };
-
-  componentDidMount() {
-    this.getTopVideosList();
-  }
+export const HomePage = () => {
+  const [topVideosList, setTopVideosList] = useState([]);
+  const [error, setError] = useState(false);
 
-  getTopVideosList = async () => {
+  const getTopVideosList = async () => {
     const topVideos = await fetchTopVideos();
```

Comment:
rename `topVideos` to `response`, to make your code more readable and also becasue now you are not using try catch here since you're already catching the error when making api call

```
if (response.error) {
      setError(true);
    } else {	   
       setTopVideosList(response.data.items);
    }	 
  };
```